src/neosophia/llmtools/tools.py

```diff
@@ -0,0 +1,32 @@
+"""
+A collection of tools that LLMs can use to solve problems.
+
+A tool is a function with a standard description.
+"""
+from collections.abc import Callable
+import sqlite3
+
+
+def make_sqlite_query_tool(db_connection: sqlite3.Connection) -> Callable:
+    """Run a sqlite query against a given database connection."""
+
+
+    description = 'Run a sqlite query against a databse.'
+    params = {
+        'query': {
+            'description': 'The query to run',
+            'typ': str,
+            'required': True
+        }
+    }
+
+
+    def sqlite_tool(query: str) -> str:
+        cursor = db_connection.cursor()
+        try:
+            results = str(cursor.execute(query).fetchall())
```

Comment:
We might want to experiment with different methods here for converting query results to a string. For our demo, Cam and I were doing the query with pandas and converting the resulting dataframe to a string, which gets a more human-readable result. I wonder if the LLM would have an easier time interpreting that.